src/discord/index.ts

```diff
@@ -12,3 +13,32 @@ export function init(): void {
 client.on('ready', () => {
     console.log(`Logged in as ${client.user!.tag}`);
 });
+
+//Temporary method to run slash commands until event handlers are set up
+client.on('interactionCreate', async (interaction) => {
+    if (!interaction.isCommand()) return;
+
+    const command = commands.find((c) => c.name === interaction.commandName);
+
+    if (command) {
+        command.execute(interaction);
+    }
+});
+
+//Temporary method to initially set up the deploy slash command / run other message commands until event handlers are set up
+client.on('messageCreate', async (message) => {
+    if (!message.content.startsWith('$') || message.author.bot) return;
+
+    const messageContent = message.content.slice(1).replace(/\s+/g, ' ').trim().split(/ (.+)/);
+    const commandName = messageContent.shift()?.toLowerCase();
+
+    if (commandName === 'deploy') {
+        const data = {
+            name: 'deploy',
+            description: 'Deploys slash commands in current guild - meant for testing.',
+            options: [],
+        };
```

Comment:
I don't think this makes a lot of sense to me - we should iterate through our array of commands, read their values (name, description, options if exists) and deploy all of them at the same time.

Reply:
Ah okay, that is what the deploy slash command does at the moment. This message command was to create the slash command on a server that didn't originally have the "/deploy" command available. If it makes more sense, I will alter the message command to deploy all the commands though (instead of having to deploy once again using the slash command). :)

Reply:
Oops, my bad - somehow I missed seeing the deploy command. I think the setup is fine then